### PR TITLE
XW-1741 | add sitename and vertical API response

### PR DESCRIPTION
**Pull request by tomasznapieralski**
https://wikia-inc.atlassian.net/browse/XW-1741

@Wikia/x-wing

### CODE REVIEW

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 1]
```diff
@@ -433,8 +435,33 @@ public function getData() {
 		return $data;
 	}
 
+	private function getSitenameData() {
+		$value = $this->wg->Sitename;
+
+		if ( $wgSitenameForComscore = WikiFactory::getVarValueByName( 'wgSitenameForComscore', $this->wikiId ) ) {
```
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: This variable name is already used in global scope. It means that it's really easy to get into a problem where we get variable names conflict. Can we rename it to reflect fact that this variable is not related to current wiki?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 2]
```diff
@@ -433,8 +435,33 @@ public function getData() {
...
+		}
+
+		return [
+			'type' => 'text',
+			'value' => $value
+		];
+	}
+
+	private function getVerticalData() {
+		$key = WikiFactoryHub::getInstance()->getWikiVertical( $this->wikiId )['short'];
```
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: `key`? Can we make it more descriptive?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 3]
```diff
@@ -381,7 +381,9 @@ public function getData() {
 					'type' => 'translatable-text',
 					'key' => 'global-footer-licensing-description',
```
Comment (Grunny) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: If the vertical is available, this message key should be `global-footer-licensing-and-vertical-description`. Though keep in mind, about 30,000 wikis don't have a vertical, so those should still use `global-footer-licensing-description`. We previously had logic for choosing which message to use based on whether the vertical was available, which was removed in https://github.com/Wikia/app/pull/10871 when we were temporarily removing the vertical while a decision was made.

Comment (tomasznapieralski) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: yes true i spot that when i was implementing this on Mercury :)

**Comment by tomasznapieralski** [comment]
https://github.com/Wikia/mercury/pull/2688

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 4]
```diff
@@ -458,7 +485,7 @@ private function getFandomOverview() {
 					'brand' => 'games',
 					'title' => [
 						'type' => 'translatable-text',
-						'key' => 'global-footer-fandom-link-vertical-games'
+						'key' => 'global-footer-vertical-games'
```
Comment (rogatty) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Please update to be compliant with https://github.com/Wikia/design-system-i18n/pull/15

**Comment by tomasznapieralski** [comment]
ready for CR

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 5]
```diff
@@ -433,8 +435,40 @@ public function getData() {
...
+		return [
+			'type' => 'text',
+			'value' => $sitename
+		];
+	}
+
+	private function getVerticalData() {
+		$verticalData = WikiFactoryHub::getInstance()->getWikiVertical( $this->wikiId );
+
+		if ($verticalData['id'] === '0') {
```
Comment (Warkot) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Spaces after `(` and before `)`. 👾

Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: What do you think about using `WikiFactoryHub::VERTICAL_ID_OTHER`?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 6]
```diff
@@ -433,8 +435,40 @@ public function getData() {
...
+			'type' => 'text',
+			'value' => $sitename
+		];
+	}
+
+	private function getVerticalData() {
+		$verticalData = WikiFactoryHub::getInstance()->getWikiVertical( $this->wikiId );
+
+		if ( $verticalData['id'] == WikiFactoryHub::VERTICAL_ID_OTHER ) {
+			$verticalMessageKey = 'lifestyle';
```
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: Please fetch this key from `WikiFactoryHub` or extract it to class constant.

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 7]
```diff
@@ -433,8 +435,40 @@ public function getData() {
 		return $data;
 	}
 
+	private function getSitenameData() {
+		$sitename = $this->wg->Sitename;
+
+		if ( $wgSitenameForComscoreForWikiId = WikiFactory::getVarValueByName( 'wgSitenameForComscore', $this->wikiId ) ) {
+			$sitename = $wgSitenameForComscoreForWikiId;
+		} else if ( $wgSitenameForWikiId = WikiFactory::getVarValueByName( 'wgSitename', $this->wikiId ) ) {
```
Comment (Warkot) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: `elseif`

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 8]
```diff
@@ -433,8 +435,40 @@ public function getData() {
 		return $data;
 	}
 
+	private function getSitenameData() {
+		$sitename = $this->wg->Sitename;
+
+		if ( $wgSitenameForComscoreForWikiId = WikiFactory::getVarValueByName( 'wgSitenameForComscore', $this->wikiId ) ) {
```
Comment (Warkot) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: It's not the nicest way to do the comparison by writing an assignment in condition. 😕

Comment (rogatty) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP

> Using assignment as an expression is surprising to the reader and looks like an error. Do not write code like this

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 9]
```diff
@@ -433,8 +435,46 @@ public function getData() {
...
+			$wgSitenameForWikiId = WikiFactory::getVarValueByName( 'wgSitename', $this->wikiId );
+
+			if ( $wgSitenameForWikiId ) {
+				$sitename = $wgSitenameForWikiId;
+			}
+		}
+
+		return [
+			'type' => 'text',
+			'value' => $sitename
```
Comment (hakubo) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: What do you thin about?
$sitename ?? $this->wg->Sitename

I think the flow here would be cleaner

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 10]
```diff
@@ -433,8 +435,46 @@ public function getData() {
...
+		$wikiFactoryInstance = WikiFactoryHub::getInstance();
+		$verticalData = $wikiFactoryInstance->getWikiVertical( $this->wikiId );
+
+		if ( $verticalData['id'] == WikiFactoryHub::VERTICAL_ID_OTHER ) {
+			$verticalMessageKey = $wikiFactoryInstance->getAllVerticals()[WikiFactoryHub::VERTICAL_ID_LIFESTYLE]['short'];
+		} else {
+			$verticalMessageKey = $verticalData['short'];
+		}
+
+		$verticalMessageKey = 'global-footer-licensing-and-vertical-description-param-vertical-' . $verticalMessageKey;
```
Comment (hakubo) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:483]: these are hard to find when needed
can you list all the options in a comment?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 11]
```diff
@@ -433,8 +435,46 @@ public function getData() {
...
+
+		if ( $wgSitenameForComscoreForWikiId ) {
+			$sitename = $wgSitenameForComscoreForWikiId;
+		} else {
+			$wgSitenameForWikiId = WikiFactory::getVarValueByName( 'wgSitename', $this->wikiId );
+
+			if ( $wgSitenameForWikiId ) {
+				$sitename = $wgSitenameForWikiId;
+			}
+		}
```
Comment (kvas-damian) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:451]: What do you think about using 4th parameter (`$default`) in `getVarValueByName()` to simplify the code?

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 12]
```diff
@@ -436,18 +436,22 @@ public function getData() {
...
-			$sitename = $wgSitenameForComscoreForWikiId;
-		} else {
-			$wgSitenameForWikiId = WikiFactory::getVarValueByName( 'wgSitename', $this->wikiId );
-
-			if ( $wgSitenameForWikiId ) {
-				$sitename = $wgSitenameForWikiId;
-			}
-		}
+		/**
+		 * take value from wgSitenameForComscore if set
```
Comment (hakubo) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php]: to me this is redundant with code

File: includes/wikia/models/DesignSystemGlobalFooterModel.class.php [hunk 13]
```diff
@@ -459,6 +463,10 @@ private function getVerticalData() {
 		$wikiFactoryInstance = WikiFactoryHub::getInstance();
 		$verticalData = $wikiFactoryInstance->getWikiVertical( $this->wikiId );
 
+		/**
+		 * We don't want to show vertical 'Other' instead we show vertical 'Lifestyle'
+		 * This is Comscore requirement
```
Comment (hakubo) [inline includes/wikia/models/DesignSystemGlobalFooterModel.class.php:465]: this is a good comment - 1st line is 'obvious' from code
but the second line nails it - that is a context that is very valuable - but cannot be read from code

**Comment by hakubo** [comment]
LGTM!

**Comment by kvas-damian** [comment]
LGTM
